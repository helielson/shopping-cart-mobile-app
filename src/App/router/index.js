/**
 * @format
 * @flow
 */
import React from 'react';
import { createAppContainer } from 'react-navigation';
import { createMaterialBottomTabNavigator } from 'react-navigation-material-bottom-tabs';
import Icon from 'react-native-vector-icons/MaterialIcons';

import ProductScreen from '../screens/Product';
import CartScreen from '../screens/Cart';
import colors from '../config/colors';

type TabBarIconProps = {
  tintColor: string,
};
const TabBarIconGenerator = name => {
  const TabBarIcon = ({ tintColor }: TabBarIconProps) => (
    <Icon name={name} size={30} color={tintColor} />
  );
  return TabBarIcon;
};

const TabNavigation = createMaterialBottomTabNavigator(
  {
    Product: {
      screen: ProductScreen,
      navigationOptions: {
        title: 'Shopping',
        tabBarIcon: TabBarIconGenerator('store'),
      },
    },
    Cart: {
      screen: CartScreen,
      navigationOptions: {
        title: 'Review your cart',
        tabBarIcon: TabBarIconGenerator('shopping-cart'),
      },
    },
  },
  {
    initialRouteName: 'Product',
    barStyle: {
      backgroundColor: colors.ternary,
    },
    activeColor: colors.backgroundLight,
  }
);

export default createAppContainer(TabNavigation);
