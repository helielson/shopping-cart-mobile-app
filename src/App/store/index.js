/**
 * @format
 * @flow
 */

import { createStore, applyMiddleware } from 'redux';
import { connect } from 'react-redux';
import logger from 'redux-logger';
import {
  createReduxContainer,
  createReactNavigationReduxMiddleware,
} from 'react-navigation-redux-helpers';
import thunkMiddleware from 'redux-thunk';

import AppNavigator from '../router';

// import new redux helpers

import appReducer from './reducers';

const middleware = createReactNavigationReduxMiddleware(state => state.nav);

const App = createReduxContainer(AppNavigator);

const mapStateToProps = state => ({
  state: state.nav,
});

export const AppWithNavigationState = connect(mapStateToProps)(App);

export default createStore(appReducer, applyMiddleware(thunkMiddleware, logger, middleware));
