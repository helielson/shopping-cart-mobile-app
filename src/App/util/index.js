/**
 * @format
 * @flow
 */

/**
 * Format number to price in Real
 * (Ex:20300.14908 -> R$ 20.300,15)
 *
 * @param {number} money amount of money
 * @returns {string} formatted value in Real
 */
export default function(money: number): string {
  return `R$ ${money
    .toFixed(2)
    .replace('.', ',')
    .replace(/(\d)(?=(\d{3})+,)/g, '$1.')}`;
}
