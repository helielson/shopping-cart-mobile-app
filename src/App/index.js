/**
 * @format
 * @flow
 */
import React from 'react';
import { Provider } from 'react-redux';
import store, { AppWithNavigationState } from './store';

const App = () => (
  <Provider store={store}>
    <AppWithNavigationState />
  </Provider>
);

export default App;
