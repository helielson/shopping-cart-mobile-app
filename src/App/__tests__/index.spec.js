/** @format */

import React from 'react';
import { shallow } from 'enzyme';

import App from '..';

describe('<App />', () => {
  test('should render properly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
