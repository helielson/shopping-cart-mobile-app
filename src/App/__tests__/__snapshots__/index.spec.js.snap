// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "fontSize": 16,
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Your App is ready and you already have:
  </Text>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginVertical": 5,
        }
      }
    >
      ✅ ESLint
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginVertical": 5,
        }
      }
    >
      ✅ Prettier
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginVertical": 5,
        }
      }
    >
      ✅ Lint Staged
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginVertical": 5,
        }
      }
    >
      ✅ Flow
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginVertical": 5,
        }
      }
    >
      ✅ Custom Scripts
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#333333",
        "fontSize": 16,
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    To get started, edit App.js
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "fontSize": 16,
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Press Cmd+R to reload,
Cmd+D or shake for dev menu
  </Text>
</View>
`;
